tests/Feature/Controllers/Api/ProjectControllerTest.php

```diff
@@ -0,0 +1,566 @@
+            'title' => 'Another title',
+            'description' => 'The description has been changed.',
+            'status' => 'pending',
+            'due_date' => now()->addMonths(4)->format('d-m-Y'),
+            'manager_id' => $this->manager->id,
+        ];
+
+        $project = Project::factory()->create(['due_date' => now()->addMonths(6)]);
+
+        $this->actingAs($this->manager)->putJson('/api/v1/projects/'.$project->id, $data)
+            ->assertOk();
+
+        $this->assertEquals($project->fresh()->title, $data['title']);
+        $this->assertEquals($project->fresh()->description, $data['description']);
+        $this->assertEquals($project->fresh()->status, $data['status']);
+        $this->assertEquals($project->fresh()->due_date->format('d-m-Y'), $data['due_date']);
+        $this->assertEquals($project->fresh()->manager_id, $data['manager_id']);
+    }
+
+    public function test_a_user_without_the_pin_project_permission_cannot_pin_a_project()
+    {
+        $data = [
+            'title' => 'Another title',
+            'description' => 'The description has been changed.',
+            'status' => 'pending',
+            'due_date' => now()->addMonths(4)->format('d-m-Y'),
+            'manager_id' => $this->manager->id,
+            'is_pinned' => true,
+        ];
+
+        $project = Project::factory()->create(['due_date' => now()->addMonths(6)]);
+
+        $this->actingAs($this->manager)->putJson('/api/v1/projects/'.$project->id, $data)
+            ->assertForbidden()
+            ->assertJsonFragment([
+                'error' => 'Not authorized',
+                'message' => 'You are not authorized to pin a project.',
+            ]);
+
+        $this->assertNotEquals($project->fresh()->is_pinned, $data['is_pinned']);
+    }
+
+    public function test_an_admin_cannot_pin_another_project_when_a_project_has_already_been_pinned()
+    {
+        $data = [
+            'title' => 'Another title',
+            'description' => 'The description has been changed.',
+            'status' => 'pending',
+            'due_date' => now()->addMonths(4)->format('d-m-Y'),
+            'manager_id' => $this->manager->id,
+            'is_pinned' => true,
+        ];
+
+        Project::factory()->create([
+            'due_date' => now()->addMonths(4),
+            'is_pinned' => true,
+        ]);
+
+        $project = Project::factory()->create(['due_date' => now()->addMonths(6)]);
+
+        $this->actingAs($this->admin)->putJson('/api/v1/projects/'.$project->id, $data)
+            ->assertUnprocessable()
+            ->assertJsonFragment([
+                'error' => 'Invalid pinned project',
+                'message' => 'There is a pinned project already. If you want to pin this project you will have to unpin the other project.',
+            ]);
+
+        $this->assertNotEquals($project->fresh()->is_pinned, $data['is_pinned']);
+    }
+
+    public function test_a_guest_cannot_access_the_destroy_project_endpoint()
```

Comment:
in deze en opvolgende tests wordt er gebruikt gemaakt van 2 termen voor hetzelfde endpoint. Naar een term gaan